python/tank/platform/software_launcher.py

```diff
@@ -336,47 +375,83 @@ def get_standard_plugin_environment(self):
 
         return env
 
-    def is_version_supported(self, version):
+    def is_version_supported(self, sw_version):
```

Comment:
I think this is part of a bigger discussion. Right now `scan_software` returns `SoftwareVersion` objects that, presumably knows everything about the executable they represent. And we're currently doing the `is_version_supported()` check in `scan_software`. It feels like the launcher really should be handling the filtering (to me at least) in which case it will only have access to the `SoftwareVersion`. If you want to override the filtering, you implement `is_version_supported()`. I'd rather lock that interface down to a single object argument than limit it to strings, especially if we move the default filtering up to the launcher. I'm also thinking we can generalize the SW scanning, but again, that's a bigger discussion.